scripts/browse.py

```diff
@@ -80,36 +82,72 @@ def create_message(chunk, question):
         "content": f"\"\"\"{chunk}\"\"\" Using the above text, please answer the following question: \"{question}\" -- if the question cannot be answered using the text, please summarize the text."
     }
 
-def summarize_text(text, question):
-    if not text:
-        return "Error: No text to summarize"
 
-    text_length = len(text)
-    print(f"Text length: {text_length} characters")
+@retry(wait=wait_exponential(multiplier=1, min=4, max=30), stop=stop_after_attempt(5))
+def call_create_chat_completion(messages):
+    return create_chat_completion(
+        model=cfg.fast_llm_model,
+        messages=messages,
+        max_tokens=300,
+
 
+def summarize_text(text, is_website=True):
+    if text == "":
+        return "Error: No text to summarize"
+
+    print("Text length: " + str(len(text)) + " characters")
     summaries = []
     chunks = list(split_text(text))
 
     for i, chunk in enumerate(chunks):
-        print(f"Summarizing chunk {i + 1} / {len(chunks)}")
-        messages = [create_message(chunk, question)]
-
-        summary = create_chat_completion(
-            model=cfg.fast_llm_model,
-            messages=messages,
-            max_tokens=300,
-        )
-        summaries.append(summary)
-
-    print(f"Summarized {len(chunks)} chunks.")
+        print("Summarizing chunk " + str(i + 1) + " / " + str(len(chunks)))
+        if is_website:
+            messages = [
+                {
+                    "role": "user",
+                    "content": "Please summarize the following website text, do not describe the general website, but instead concisely extract the specific information this subpage contains.: " +
+                    chunk},
+            ]
+        else:
+            messages = [
```

Comment:
This stuff doesn't follow the design pattern in this file. Currently this kind of material exists in methods, for example `create_message`. I would move this logic up into the class as a method(s) and try to keep the implementation in this part of the file as similar to the original in form and function as possible, except for calling a different method.

Reply:
updated.